handsontable/src/plugins/filters/filters.js

```diff
@@ -857,11 +857,18 @@ export class Filters extends BasePlugin {
    *
    * @param {number} col Visual column index.
    * @param {HTMLTableCellElement} TH Header's TH element.
+   * @param {number} headerLevel The index of header level counting from the top (positive
+   *                             values counting from 0 to N).
+   *
    */
-  #onAfterGetColHeader(col, TH) {
+  #onAfterGetColHeader(col, TH, headerLevel) {
     const physicalColumn = this.hot.toPhysicalColumn(col);
 
-    if (this.enabled && this.conditionCollection.hasConditions(physicalColumn)) {
+    if (
+      this.enabled
+      && this.conditionCollection.hasConditions(physicalColumn)
+      && headerLevel === (this.hot.getSettings().nestedHeaders?.length || 1) - 1
+    ) {
```

Comment:
This won't work for nested headers added by non-plugin code (e.g. custom user implementation). I think this change should work for all cases
```suggestion
    if (
      this.enabled
      && this.conditionCollection.hasConditions(physicalColumn)
      && headerLevel === this.hot.view.getColumnHeadersCount() - 1
    ) {
```